### PR TITLE
be much more quiet on success.  

### PR DESCRIPTION
**Pull request by osheroff**
Capture the output of the child via pipes and eat it on success, unless `--verbose` is specified.

@grosser

### CODE REVIEW

**Comment by osheroff** [comment]
ping @grosser

File: lib/forking_test_runner.rb [hunk 1]
```diff
@@ -146,8 +157,14 @@ def run_test(file)
...
         ActiveRecord::Base.connection.disconnect!
       end
+      output = nil
+
       change_program_name_to file do
+        rpipe, wpipe = IO.pipe
+
         child = fork do
+          rpipe.close
+          $stdout.reopen(wpipe)
```
Comment (grosser) [inline lib/forking_test_runner.rb]: this will suppress all output until the test is complete, can we only do this if we are not in verbose mode ?

**Comment by grosser** [comment]
This --verbose mode is a good default for any new setup / while trying to debug + all the tests still use it ... how about --quiet instead ?

**Comment by grosser** [comment]
some docs on the flag and at least 1 test hitting that codepath would be nice :)

**Comment by osheroff** [comment]
yeah `--quiet` is better.  damnit.

**Comment by osheroff** [comment]
addressed.

File: lib/forking_test_runner.rb [hunk 2]
```diff
@@ -146,8 +159,17 @@ def run_test(file)
         preload_fixtures
         ActiveRecord::Base.connection.disconnect!
       end
+      output = nil
+
       change_program_name_to file do
+        rpipe, wpipe = IO.pipe
```
Comment (grosser) [inline lib/forking_test_runner.rb]: `if @verbose` to not leave a pipe hanging ?

File: lib/forking_test_runner.rb [hunk 3]
```diff
@@ -16,17 +20,26 @@ def cli(argv)
...
+
+        puts "Time: expected #{expected.round(2)}, actual #{time.round(2)}" if runtime_log && @verbose
+        puts "#{CLEAR} <<< #{file} ---- #{success ? "OK" : "Failed"}" if @verbose
         [file, time, expected, success]
       end
 
+      puts
       # pretty print the results
       puts "\nResults:"
       puts results.
```
Comment (grosser) [inline lib/forking_test_runner.rb]: hide results when quiet / just duplicates the previous output

File: lib/forking_test_runner.rb [hunk 4]
```diff
@@ -56,7 +69,7 @@ def benchmark
       time = Benchmark.realtime do
         result = yield
       end
-      return time, result
+      return [time, result].flatten
```
Comment (grosser) [inline lib/forking_test_runner.rb]: why does it need flatten now ?

**Comment by grosser** [comment]
:+1:

File: lib/forking_test_runner.rb (hunk 4 above)
Comment (osheroff) [inline lib/forking_test_runner.rb]: i wanted to return `[time, success, output]` not `[time, [success, output]]`

File: lib/forking_test_runner.rb (hunk 3 above)
Comment (osheroff) [inline lib/forking_test_runner.rb]: no idea what you mean with this comment

**Comment by osheroff** [comment]
refactored the output capture.  much cleaner now.

**Comment by osheroff** [comment]
@grosser merge & release?

**Comment by grosser** [comment]
0.5.0

refactored a bit more and added the result summarizer

**Pull request merged**